### DISCUSSION TITLE
zowe create dataset pds error

### DISCUSSION THREAD

**Discussion by jdrt77** [zowe/zowe-cli#1592] [General]
Hi there!!

I'm getting the next error when i try to create a new PDS through Zowe CLI o VSCode

command:
zowe files create pds --data-class spds 'Z27436.CBL'

error:
Command Error:
z/OSMF REST API Error:
Rest API failure with HTTP(S) status 500
category: 8
rc:       -26868
reason:   0
message:  Dynamic allocation Error

Error Details:
HTTP(S) error status "500" received.
Review request details (resource, base path, credentials, payload) and ensure correctness.

Protocol:  https
Host:      204.90.115.200
Port:      10443
Base Path:
Resource:  /zosmf/restfiles/ds/z27436.cbl
Request:   POST
Headers:   [{"Accept-Encoding":"gzip"},{"X-CSRF-ZOSMF-HEADER":true}]
Payload:   '{"alcunit":"CYL","dsorg":"PO","primary":1,"dirblk":5,"recfm":"FB","blksize":6160,"lrecl":80,"secondary":1}'

I've tried to create the library putting all the values like lrecl, recfm, ... but i get the same error

Could you help me, please??

Thanks a lot!!

**Comment by zFernand0** [comment]
Hey @jdrt77,
I'm not sure what the configuration on that system is, but is the `--data-class spds` a valid class?
```
# zowe files create pds TESTDS.TESTDS --data-class spds
Command Error:
z/OSMF REST API Error:
Rest API failure with HTTP(S) status 500
category: 8
rc:       -26880
reason:   0
message:  Dynamic allocation Error - SMS or ACS services returned an error.  
Ensure any specified storclas, dataclas, or mgmtclas values are defined.
...
```

Also, if the dataset already exist you will get the same error but with less details
```
# zowe files create pds TESTDS.TESTDS
Command Error:
z/OSMF REST API Error:
Rest API failure with HTTP(S) status 500
category: 8
rc:       -26868
reason:   0
message:  Dynamic allocation Error
...
```
